src/Umbraco.Cms.Integrations.Search.Algolia/Services/ScopeService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Umbraco.Cms.Infrastructure.Scoping;
+using Umbraco.Cms.Integrations.Search.Algolia.Migrations;
+
+namespace Umbraco.Cms.Integrations.Search.Algolia.Services
+{
+    public class ScopeService : IScopeService<AlgoliaIndex>
+    {
+        private readonly IScopeProvider _scopeProvider;
+
+        public ScopeService(IScopeProvider scopeProvider)
+        {
+            _scopeProvider = scopeProvider;
+        }
+
+        public void AddOrUpdate(AlgoliaIndex entity)
+        {
+            using var scope = _scopeProvider.CreateScope();
+
+            if (entity.Id == 0)
+                scope.Database.Insert(entity);
+            else
+                scope.Database.Update(entity);
+
+            scope.Complete();
+        }
+
+        public List<AlgoliaIndex> Get()
+        {
+            using var scope = _scopeProvider.CreateScope();
+
+            return scope.Database.Fetch<AlgoliaIndex>();
+        }
+
+
+        public AlgoliaIndex GetById(int id)
+        {
+            using var scope = _scopeProvider.CreateScope();
+
+            return scope.Database.Single<AlgoliaIndex>("where Id = " + id);
+        }
+
+        public List<AlgoliaIndex> GetByContentTypeAlias(string alias)
+        {
+            using var scope = _scopeProvider.CreateScope();
+
+            return scope.Database.Fetch<AlgoliaIndex>("where SerializedData like '%" + alias + "%'");
```

Comment:
This seems vulnerable to errors - could be that the property aliases and content type aliases match so you'd get false returns here.  Maybe you need to think about storing content types in a related table if you need to be able to get back all indexes that include a particular content type, or review if you really need this method.